chall06/edramire.c

```diff
@@ -0,0 +1,116 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   edramire.c                                         :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: edramire <edramire@student.42.fr>          +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2020/04/20 11:22:35 by edramire          #+#    #+#             */
+/*   Updated: 2020/04/20 21:19:48 by edramire         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include <stdio.h>
+#include <string.h>
+
+static int	get_chess_size(char *board, int len)
+{
+	int		size;
+	int		row;
+	int		col;
+
+	if (len == 0 || len > 89)
+		return (0);
+	size = 0;
+	row = 0;
+	while (*board != '\0')
+	{
+		col = 0;
+		while (*board != '\0' && *board != '\n')
+		{
+			++col;
+			++board;
+		}
+		if (row == 0)
+			size = col;
+		else if (col != size)
+			return (0);
+		++row;
+		board += *board == '\n' ? 1 : 0;
+	}
+	return (row == col ? size : 0);
```

Comment:
You successfully return `0` for a `\n` at the beginning of the board, but not at the end. (But this is a bit of an edge case and it wasn't completely specified)

Reply:
I checked the error commented by you, but that function is only to get the size of the line. In case the line has a new line character at the beginning the funcion check_mate will return 1, as specified in the subject of the challenge.

Reply:
Yes you check for "\n...K\n...R....\n...." ==  invalid cause `\n` at the beginning, but what about "...K\n...R....\n....\n" == invalid cause `\n` at the end ? 

Anyway I'm not sure this tests will be included in the end.

Reply:
In that case, the function will fail on a 9x9 map with new line character at the end, i take only as valid a string with 81 character and 8 new line character; this validation are incomplete 
`if (len == 0 || len > 89)`. On smaller maps has will not problem

Reply:
If the map is already 9x9 it will detect it yeah but for example here -> `board = ".R.....K\n........\n........\n........\n........\n........\n........\n........\n"` it's an 8x8 with an extra `\n` at the end it should return `1` cause of it, but you return `0`. Anyway, again, I don't think we will add this test at the end. :)